community/modules/scripts/wait-for-startup/main.tf

```diff
@@ -13,22 +13,22 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-data "google_compute_instance" "vm_instance" {
-  name    = var.instance_name
-  zone    = var.zone
-  project = var.project_id
-}
+# data "google_compute_instance" "vm_instance" {
+#   name    = var.instance_name
+#   zone    = var.zone
+#   project = var.project_id
+# }
 
 resource "null_resource" "wait_for_startup" {
   provisioner "local-exec" {
     command = "/bin/bash ${path.module}/scripts/wait-for-startup-status.sh"
     environment = {
-      INSTANCE_NAME = var.instance_name
-      ZONE          = var.zone
-      PROJECT_ID    = var.project_id
-      TIMEOUT       = var.timeout
+      INSTANCE_NAMES = join(" ", distinct(compact(concat(var.instance_names, [var.instance_name]))))
+      ZONE           = var.zone
```

Comment:
While it's not directly related to this change, 
but using `zone` and `project` as separate inputs can be error-prone.E.g. was created 
Better option would be to use self-link. 

#2460 adds `self_link` to the output.

If you intend to use `data "google_compute_instance" "vm_instance"`  you better make use of returned `self_link`.

Reply:
I don't think we can use a "data" block with triggers because of the issues we found previously with modifying lists in Terraform (this is why we agreed to move all the complexity into the script).

Maybe in the future, we can add a variable for `self_links` so that `wait-for-startup` can "use" `vm-instance`.  Unfortunately, it seems like `gcloud`'s  `get-serial-output` subcommand doesn't have an argument for self_link, so we'd probably need to parse the zone and project out of the self_link.

Reply:
> because of the issues we found previously with modifying lists in Terraform

I don't see why we need any list modifications here.

> Unfortunately, it seems like gcloud's get-serial-output subcommand doesn't have an argument for self_link

`$ gcloud compute instances get-serial-port-output <self_link>` - works like a charm

Reply:
Simply speaking that would be an implementation:

```hcl
data  "google_compute_instance" "instance_by_link" {
  # for each `var.self_links`
}

data  "google_compute_instance" "instance_by_name" {
  # for each `concat(var.instance_names, [var.instance_name])`
}

resource "null_resource" "wait_for_startup" {
   INSTANCES_NAMES = join(" ", distinct(concat(instance_by_link[*].self_link,  instance_by_name[*].self_link)))
}

```

Do you see any problems here ?

Reply:
> We can't use for_each

Sure, since it's a list, just use `count` instead.

Reply:
Summary from offline conversation:
1. We will try to merge this change since it provides immediate value to other projects.
2. We need to come up with a plan to take a list of "self_link" values as an input to this module, then remove "name", "project" and "zone" inputs to improve the experience when using zone-finding or multiple zones.